### PR TITLE
Some fixes to `send_event` and `send_chunk` extension APIs

### PR DESCRIPTION
**Pull request by aleclarson**
- Correctly retain the context when:
  - Transfer-Encoding header is set to `chunked`
  - Content-Type header is set to `text/event-stream`
- Add debug log for incorrect use (failure to set expected header)

### CODE REVIEW

**Review by jwerle — APPROVED** [review]

File: src/extension/ipc.cc [hunk 1]
```diff
@@ -137,8 +137,7 @@ bool sapi_ipc_reply (const sapi_ipc_result_t* result) {
   }
 
   // if retained, then then caller must eventually call `sapi_context_release()`
-  if (!context->retained) {
-    context->release();
+  if (context->release()) {
```
Comment (aleclarson) [inline src/extension/ipc.cc:140]: Ah wait, this doesn't seem right

Comment (aleclarson) [reply src/extension/ipc.cc:138]: For this to work, we have to allocate the route context for each message. See my latest commit.

**Comment by jwerle** [comment]
@aleclarson LGTM - is this ready?

**Comment by aleclarson** [comment]
👍

**Pull request merged**